src/topology/uniform_space/separation.lean

```diff
@@ -2,44 +2,106 @@
 Copyright (c) 2017 Johannes Hölzl. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Johannes Hölzl, Patrick Massot
-
-Hausdorff properties of uniform spaces. Separation quotient.
 -/
+
 import topology.uniform_space.basic
+import tactic.apply_fun
+
+/-!
+# Hausdorff properties of uniform spaces. Separation quotient.
+
+This file studies uniform spaces whose underlying topological spaces are separated
+(also known as Hausdorff or T₂).
+This turns out to be equivalent to asking that the intersection of all entourages
+is the diagonal only. This conditions actually implies the stronger separation property
+that the space is regular (T₃), hence those conditions are equivalent for topologies coming from
+a uniform structure.
+
+More generally, the intersection `𝓢 X` of all entourages of `X`, which has type `set (X × X)` is an
+equivalence relation on `X`. Points which are equivalent under the relation are basically
+undistinguishable from the point of view of the uniform structure. For instance any uniformly
+continuous function will send equivalent points to the same value.
+
+The quotient `separation_quotient X` of `X` by `𝓢 X` has a natural uniform structure which is
+separated space, and satisfies a universal property: every uniformly continuous function
+from `X` to a separated uniform space uniquely factors through `separation_quotient X`.
+As usual, this allows to turn `separation_quotient` into a functor (but we don't use the
+category theory library in this file).
+
+These notions admit relative versions, one can ask that `s : set X` is separated, this
+is equivalent to asking that the uniform structure induced on `s` is separated.
+
+## Main definitions
+
+* `separation_relation X : set (X × X)`: the separation relation
+* `separated X`: a predicate class asserting that `X` is separated
```

Comment:
The usual classes are postfixed with `space` (like `complete_space`, `compact_space`, `proper_space` and so on), while the set versions are prefixed with `is` (like `is_open`, `is_complete`, `is_compact` and so on). Do you think it would be possible to adhere to this general convention also here (with `separated_space` and `is_separated` respectively)?

Reply:
I renamed.